### ISSUE TITLE
External monitor plug crashes kitty

### ISSUE THREAD

**Issue by davc0n** [hyprwm/Hyprland#6510]
### Bug or Regression?

Bug

### System Info and Version

<details> <summary>System/Version info</summary>

```sh
Hyprland, built from branch  at commit 9e781040d9067c2711ec2e9f5b47b76ef70762b3  (props: bump version to 0.41.1).
Date: Thu Jun 13 09:54:06 2024
Tag: v0.41.1, commits: 4818

flags: (if any)


System Information:
System name: Linux
Node name: thinkpad
Release: 6.9.4-arch1-1
Version: #1 SMP PREEMPT_DYNAMIC Wed, 12 Jun 2024 20:17:17 +0000


GPU information: 
03:00.0 VGA compatible controller [0300]: Advanced Micro Devices, Inc. [AMD/ATI] Picasso/Raven 2 [Radeon Vega Series / Radeon Vega Mobile Series] [1002:15d8] (rev d2) (prog-if 00 [VGA controller])


os-release: NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://gitlab.archlinux.org/groups/archlinux/-/issues"
PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
LOGO=archlinux-logo


plugins:

======Config-Start======


Config File: /home/davide/.config/hypr/hyprland.conf: Read Succeeded
monitor=eDP-1,preferred,1920x1080,1.2
monitor=HDMI-A-1,preferred,0x0,1
monitor=,preferred,auto,auto

exec-once = waybar
exec-once = wl-paste --type text --watch cliphist store
exec-once = wl-paste --type image --watch cliphist store

$terminal = kitty --single-instance
$menu = wofi --gtk-dark --show drun --define key_expand=space

env = XCURSOR_SIZE,24
env = QT_QPA_PLATFORMTHEME,qt5ct # change to qt6ct if you have that

xwayland {
    force_zero_scaling = true
}

# For all categories, see https://wiki.hyprland.org/Configuring/Variables/
input {
    kb_layout = it
    kb_options = caps:escape

    follow_mouse = 1

    touchpad {
        natural_scroll = no
    }

    sensitivity = 0 # -1.0 to 1.0, 0 means no modification.
}

general {
    # See https://wiki.hyprland.org/Configuring/Variables/ for more

    gaps_in = 0 
    gaps_out = 0 
    col.active_border = rgba(33ccffee) rgba(00ff99ee) 45deg
    col.inactive_border = rgba(595959aa)

    layout = dwindle

    # Please see https://wiki.hyprland.org/Configuring/Tearing/ before you turn this on
    allow_tearing = false
}

decoration {
    # See https://wiki.hyprland.org/Configuring/Variables/ for more

    blur {
        enabled = true
        size = 3
        passes = 1
    }

    drop_shadow = yes
    shadow_range = 4
    shadow_render_power = 3
    col.shadow = rgba(1a1a1aee)
}

animations {
    enabled = yes

    # Some default animations, see https://wiki.hyprland.org/Configuring/Animations/ for more

    bezier = myBezier, 0.05, 0.9, 0.1, 1.05

    animation = windows, 1, 7, myBezier
    animation = windowsOut, 1, 7, default, popin 80%
    animation = border, 1, 10, default
    animation = borderangle, 1, 8, default
    animation = fade, 1, 7, default
    animation = workspaces, 1, 6, default
}

dwindle {
    # See https://wiki.hyprland.org/Configuring/Dwindle-Layout/ for more
    pseudotile = yes # master switch for pseudotiling. Enabling is bound to mainMod + P in the keybinds section below
    preserve_split = yes # you probably want this
}

master {
    # See https://wiki.hyprland.org/Configuring/Master-Layout/ for more
    new_is_master = true
}

gestures {
    # See https://wiki.hyprland.org/Configuring/Variables/ for more
    workspace_swipe = on
}

misc {
    disable_autoreload = true
}

# Example windowrule v1
# windowrule = float, ^(kitty)$
# Example windowrule v2
# windowrulev2 = float,class:^(kitty)$,title:^(kitty)$
# See https://wiki.hyprland.org/Configuring/Window-Rules/ for more
windowrulev2 = suppressevent maximize, class:.* # You'll probably like this. 

# See https://wiki.hyprland.org/Configuring/Keywords/ for more
$mainMod = SUPER

# Example binds, see https://wiki.hyprland.org/Configuring/Binds/ for more
bind = $mainMod, Q, exec, $terminal
bind = $mainMod, C, killactive,
# bind = $mainMod, V, togglefloating,
bind = $mainMod, R, exec, $menu
# bind = $mainMod, P, pseudo, # dwindle
bind = $mainMod, J, togglesplit, # dwindle

# Move focus with mainMod + arrow keys
bind = $mainMod, left, movefocus, l
bind = $mainMod, right, movefocus, r
bind = $mainMod, up, movefocus, u
bind = $mainMod, down, movefocus, d

# Switch workspaces with mainMod + [0-9]
bind = $mainMod, 1, workspace, 1
bind = $mainMod, 2, workspace, 2
bind = $mainMod, 3, workspace, 3
bind = $mainMod, 4, workspace, 4
bind = $mainMod, 5, workspace, 5
bind = $mainMod, 6, workspace, 6
bind = $mainMod, 7, workspace, 7
bind = $mainMod, 8, workspace, 8
bind = $mainMod, 9, workspace, 9
bind = $mainMod, 0, workspace, 10

# Move active window to a workspace with mainMod + SHIFT + [0-9]
bind = $mainMod SHIFT, 1, movetoworkspace, 1
bind = $mainMod SHIFT, 2, movetoworkspace, 2
bind = $mainMod SHIFT, 3, movetoworkspace, 3
bind = $mainMod SHIFT, 4, movetoworkspace, 4
bind = $mainMod SHIFT, 5, movetoworkspace, 5
bind = $mainMod SHIFT, 6, movetoworkspace, 6
bind = $mainMod SHIFT, 7, movetoworkspace, 7
bind = $mainMod SHIFT, 8, movetoworkspace, 8
bind = $mainMod SHIFT, 9, movetoworkspace, 9
bind = $mainMod SHIFT, 0, movetoworkspace, 10

# Example special workspace (scratchpad)
bind = $mainMod, S, togglespecialworkspace, magic
bind = $mainMod SHIFT, S, movetoworkspace, special:magic

# Scroll through existing workspaces with mainMod + scroll
bind = $mainMod, mouse_down, workspace, e+1
bind = $mainMod, mouse_up, workspace, e-1

# Move/resize windows with mainMod + LMB/RMB and dragging
bindm = $mainMod, mouse:272, movewindow
bindm = $mainMod, mouse:273, resizewindow

# Volume
binde = , XF86AudioRaiseVolume, exec, wpctl set-volume -l 1 @DEFAULT_AUDIO_SINK@ 5%+
binde = , XF86AudioLowerVolume, exec, wpctl set-volume @DEFAULT_AUDIO_SINK@ 5%-
bind = , XF86AudioMicMute, exec, wpctl set-mute @DEFAULT_AUDIO_SOURCE@ toggle
bind = , XF86AudioMute, exec, wpctl set-mute @DEFAULT_AUDIO_SINK@ toggle

# Switch to previous workspace
bind = $mainMod, TAB, workspace, previous

# Screen brightness
binde = , XF86MonBrightnessUp, exec, brightnessctl s +5%
binde = , XF86MonBrightnessDown, exec, brightnessctl s 5%-

# Move current workspace to monitor
bind = $mainMod SHIFT, page_up, movecurrentworkspacetomonitor, l
bind = $mainMod SHIFT, page_down, movecurrentworkspacetomonitor, r

# Lock
bind = $mainMod, L, exec, hyprlock

# Clipboard
bind = $mainMod, V, exec, cliphist list | wofi --gtk-dark --dmenu | cliphist decode | wl-copy

# Move active window
bind = $mainMod SHIFT, left, swapwindow, l
bind = $mainMod SHIFT, right, swapwindow, r
bind = $mainMod SHIFT, up, swapwindow, u
bind = $mainMod SHIFT, down, swapwindow, d

======Config-End========
```

</details>

### Description

ref. https://github.com/kovidgoyal/kitty/issues/7546

```
kernel: kitty[756]: segfault at 30 ip 0000715c7312bcb4 sp 00007ffed0b32008 error 4 in libwayland-client.so.0.23.0[715c7312a000+6000] likely on CPU 5 (core 2, socket 0)
```

### How to reproduce

1. Plug the external monitor
2. Move the workspace with kitty on the external monitor
3. Unplug the external monitor
4. Plug the external monitor again and kitty crashes (segfault, see above)

**Comment by MightyPlaza** [comment]
is this a regression? if so please bisect

**Comment by davc0n** [comment]
```
6967a31450441fc5605c05db6f65505dace4b263 is the first bad commit
commit 6967a31450441fc5605c05db6f65505dace4b263 (HEAD)
Author: Vaxry <43317083+vaxerski@users.noreply.github.com>
Date:   Sat Jun 8 10:07:59 2024 +0200

    wayland/core: move to new impl (#6268)
    
    * wayland/core/dmabuf: move to new impl
    
    it's the final countdown
```

**Comment by vaxerski** [comment]
`WAYLAND_DEBUG=1 kitty` log please

**Comment by davc0n** [comment]
[debug.txt](https://github.com/user-attachments/files/15854976/debug.txt)

**Comment by Agent00Ming** [comment]
> [debug.txt](https://github.com/user-attachments/files/15854976/debug.txt)

`[3614624.555] {Default Queue} wl_surface#27.enter(nil)` and probably related to what's happening in #6503

**Comment by vaxerski** [comment]
![image](https://github.com/hyprwm/Hyprland/assets/43317083/d454a3e3-a0fb-423e-9c83-c1892d5dfe02)
how is that possible? dafoq?

**Comment by davc0n** [comment]
That if block is not executed I guess: [hyprland.log](https://github.com/user-attachments/files/15856526/hyprland.log)

Off topic but vscode gets stuck at `Parsing workspace files: 958 / 2356 (40%)` and intellisense doesn't work.

**Comment by davc0n** [comment]
Fixed in v0.41.2 (maybe https://github.com/hyprwm/Hyprland/commit/cff0123ce62a1f721d3aa2ac0fb58019787a7778)
